libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -448,6 +449,33 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
                           rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
             // TODO implement joystick RC control to AP
+            const uint32_t tnow = AP_HAL::millis();
+            
+            for (uint8_t i=0; i<rx_joy_topic.axes_size; i++) {
+                const uint16_t override_data = static_cast<uint16_t>(rx_joy_topic.axes[i]);
+                const RC_Channel *c = rc().channel(i);
+                radio_min_val=c->get_radio_min();
+                radio_max_val=c->get_radio_max();
+
+
+                // Per MAVLink spec a value of UINT16_MAX means to ignore this field.
+                if (override_data != UINT16_MAX) {
+                    const uint16_t mapped_data=linear_interpolate(radio_min_val,radio_max_val,override_data,-1.0,1.0);
```

Comment:
`linear_interpolate` expects `float`s as args and returns a `float`. The resulting mapped data will need to be cast to a `uint16_t` when passed to `RC_Channels::set_override`.

Maybe rename variable to `const uint16_t override_data` and add a `static_cast<uint16_t>()` around the call to `linear_interpolate`.

Reply:
Why do we need to check for `UINT16_MAX` in the if statement? The override_data is assumed to be in the range of [-1.0,1.0]